File: src/main/java/io/github/chaosdave34/kitpvp/cosmetics/ProjectileTrail.java
package io.github.chaosdave34.kitpvp.cosmetics;

import lombok.Getter;
import org.bukkit.Location;
import org.bukkit.Material;

@Getter
public abstract class ProjectileTrail extends Cosmetic {

    public ProjectileTrail(String id, String name, int levelRequirement, Material icon) {
        super(id, name, levelRequirement, icon);
    }

    public abstract void playEffect(Location location);
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/cosmetics/ProjectileTrail.kt
package io.github.chaosdave34.kitpvp.cosmetics

import org.bukkit.Location
import org.bukkit.Material

abstract class ProjectileTrail(id: String, name: String, levelRequirement: Int, icon: Material) : Cosmetic(id, name, levelRequirement, icon) {
    abstract fun playEffect(location: Location)
}
